pkg/cloudfabric-controller/controllerframework/controller_framework.go

```diff
@@ -441,26 +437,27 @@ func (c *ControllerBase) ReportHealth(client clientset.Interface) {
 	}
 }
 
-func generateControllerName(existedInstanceMap map[string]v1.ControllerInstance) string {
-	for {
-		uid := uuid.NewUUID()
-		name := fmt.Sprintf("%s-%v", controllerInstanceNamePrefix, uid)
-		_, isExist := existedInstanceMap[name]
-		if !isExist {
-			return name
-		}
+func generateControllerName(controllerType string, existedInstanceMap map[string]v1.ControllerInstance) string {
+	cimInstanceId := GetInstanceId()
+	if cimInstanceId == "" {
+		klog.Fatalf("Controller Instance Manager not available.")
+	}
 
-		// Error
-		klog.Infof("Controller name %s conflict. Get a new one ", name)
+	name := fmt.Sprintf("%s-%v", strings.ToLower(controllerType), cimInstanceId)
+	_, isExist := existedInstanceMap[name]
+	if isExist {
+		klog.Fatalf("Controller instance name %s conflict. Need to restart process to get a new one ", name)
 	}
+
+	return name
 }
 
 // Get controller instances by controller type
 //		Return sorted controller instance list & error if any
 func listControllerInstancesByType(controllerType string) (map[string]v1.ControllerInstance, error) {
 	controllerInstanceMap := make(map[string]v1.ControllerInstance)
 
-	cim := GetControllerInstanceManager()
+	cim := GetInstanceHandler()
 	if cim == nil {
 		klog.Fatalf("Unexpected reference to uninitialized controller instance manager")
```

Comment:
the error of cim is nil is not returned.

Reply:
> the error of cim is nil is not returned.

CIM should start before this. If it is not initialized correctly, there is no fix. The process needs to be restarted.